android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/authorization/AccountRecordsDao.java

```diff
@@ -0,0 +1,99 @@
+package com.solana.mobilewalletadapter.walletlib.authorization;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteCursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteStatement;
+import android.net.Uri;
+import android.text.TextUtils;
+
+import androidx.annotation.IntRange;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class AccountRecordsDao extends DbContentProvider<AccountRecord>
```

Comment:
Summarizing my understanding: 

We're changing our "account" model from `PublicKeyRecord` -> `AccountRecord` because in MWA 2.0, the "account" model needs to be expanded to include chains/features/etc. A public key is now a part of the larger `AccountRecord` model.

Reply:
yep exactly. adding new account metadata fields to the database. 

having this already done will make it easier to migrate to multi-account too. we still need to store a list of AccountRecords for each authorization, but at least now the AccountRecord itself is there.